### ISSUE TITLE
JSDOC_ERROR: Jsdoc failed.

### ISSUE THREAD

**Issue by SRugina** [jsdoc2md/jsdoc-to-markdown#179]
Hi,

For some reason, I always get this error when running any jsdoc2md command. For example, in the [documentation](https://github.com/jsdoc2md/jsdoc-to-markdown/wiki/Cherry-picking-which-documentation-appears-in-output), it says running `jsdoc2md --json example.js` with the following example.js file should give me an ouput, but it gives me the error that is the title:

example.js:

        /**
         * Addition
         */
        function add () {}
        
        /**
         * Subtraction
         */
        function subtract () {}
        
        /**
         * Multiplication
         */
        function multiply () {}

package.json:

        {
          "name": "testjs2md",
          "version": "1.0.0",
          "description": "",
          "main": "example.js",
          "scripts": {
            "test": "echo \"Error: no test specified\" && exit 1"
          },
          "author": "",
          "license": "ISC",
          "dependencies": {
            "jsdoc": "^3.5.5",
            "jsdoc-to-markdown": "^4.0.1"
          }
        }

I use: PopOS 18.10 (basically ubuntu w/ a skin)
node version: v10.14.2
npm verison: 6.4.1
I also have jsdoc and jsdoc-to-markdown installed globally.
Works fine with jsdoc command for generating jsdoc documentation

What am I doing wrong? Thanks in advance for any help

**Comment by SRugina** [comment]
To note: works fine with jsdoc command for generating jsdoc documentation

**Comment by 75lb** [comment]
I failed to reproduce this on both macOS and Linux.. I pasted your example into file and tested these commands, which all worked normally: 

```
$ jsdoc2md issue-179.js
$ jsdoc2md --jsdoc issue-179.js
$ jsdoc2md --json issue-179.js
```

If this is still an issue, could you paste precise reproduction instructions please so I can see and debug the issue. If you want to trying debugging it yourself, it is failing [here](https://github.com/jsdoc2md/jsdoc-api/blob/master/lib/jsdoc-command.js#L113).

**Comment by 75lb** [comment]
I'm sorry, i still can't reproduce the issue - I've tested the steps above on mac and Linux, you can see a screen recording [here](https://drive.google.com/open?id=1a6Nl0cxs1Qw_mkHrLzFXiy4jJse6ozky).

I don't think this is an issue with jsdoc2md itself, particularly as you are also having issues with built-in modules like `crypto` not being found. Closing for now as cannot reproduce.

**Comment by echicken** [comment]
If it helps anyone else, I recently ran into `JSDOC_ERROR: Jsdoc failed` when using this package via CLI or its API. No other output.

The problem is that I had installed node.js as a snap on Ubuntu, and apparmor was getting in the way; lots of `apparmor="DENIED" operation="file_inherit"` messages in syslog. Ultimately related to this package's use of `child_process`.

I didn't fix my snap problem. Because it suited my purposes, I just switched to developing in a container via Visual Studio Code. Problem solved.

Others might have better luck using a non-snap install of node.
